gpii/node_modules/flowManager/test/data/browserChannel_solutionsRegistry.json

```diff
@@ -1,19 +1,19 @@
 {
     "win32": {
-        "org.chrome.cloud4chrome": {
```

Comment:
If this is testing some aspect of the real SR, it should use the real SR data.  Otherwise this should be test data that is clearly namespaced and which does not resemble the real SR data.

Reply:
These are tests of the browser channel endpoint of the flow manager, testing the ability to login/logout and then get preferences using a websocket.  As such, it uses some version of SR (also keys, prefsSafes, and device information), but it isn't testing the SR per se.  So the solutions could be completely un-real.  Can you explain why that is better?  I don't disagree; I just don't understand why using real solutions is an issue.

As for who does the TODO work, I'll take a stab at it, but if you'd rather do it, that's fine with me.

Reply:
In my own work making additions/deletions/updates to SR entries I have frequently had to hunt through and find all the little pockets of test SR data and update those as well.  Some of those pockets are quite large, which makes it harder to cleanly review things and bloats our code base.

This practice of mimicking the "real" data in my opinion contributes to our habit of using "kitchen sink" tests where we pass around bloated payloads and test them against bloated SR entries.  A "synthetic" SR entry just for the test can be designed to test specific things, and can be much smaller and easier to review.

In addition to using synthetic SR data to test specific functionality, I think the flip side is also true.  We need to clearly identify the tests that need to use "real" data and make sure that these actually use the real data.  Otherwise we might only *think* we've fixed an issue because the tests that use the real SR data passing.

Anyway, I am more than happy to address this as part of my LSR work, but obviously I think it's important and have a lot of thoughts about it.    If you want to try untangling one or two pieces to help, that'd be great.  Otherwise, a good TODO is fine.

Reply:
That makes sense, thanks.

It turns out that it is easy to create and use mock solutions, for these `BrowserChannel` tests, as well as mocks for the solutions registry, device reporter, and preferences . It also turned out that the same fully mocked data was fine for the integration tests.  (Aside: that confirms that the integration tests don't use the real solutions registries either).  Both test runs work fine with these mocks.

However (and I should have seen this coming), when running all of the node tests, there are failures in the settings payload tests (`%gpii-universal/gpii/node_modules/SolutionsRegistry/SettingsPayloadTests`).  That's because payload tests _do_ use the real solutions registries to verify that settings are well-formed. The mock preferences used here reference solutions that don't exist in the real solutions registry; hence the failures noting rogue settings.  That also explains why the "real" solutions used currently (`org.nvda-project` and `org.gnome.orca`) work both for the `BrowserChannelTests` and for the `SettingsPayloadTests`.  There is a sense of going in circles, because the rationale for fixing the `BrowserChannel` tests was that the web based solutions they relied on were no longer real solutions -- they were nonexistent.  Substituting mock solutions and preferences for them is pretty much the same thing.

Notes on how to proceed:
- move mock preferences to the test's data directory (hide them from the payload tests),
- modify things such that they are loaded into the test couch database nonetheless,
- OR, structure things such that the mock solutions registry is used by the payload tests.